R/descriptives.R

```diff
@@ -998,14 +999,30 @@ Descriptives <- function(jaspResults, dataset, options) {
 
 
 .plotMarginal <- function(column, variableName,
-                          rugs = FALSE, displayDensity = FALSE) {
+                          rugs = FALSE, displayDensity = FALSE, binWidthType = c("doane", "fd", "scott", "sturges", "manual"),
+                          numberOfBins = NA) {
+  binWidthType <- match.arg(binWidthType)
   column <- as.numeric(column)
   variable <- na.omit(column)
 
-  if(length(variable) == 0)
+  if (length(variable) == 0)
     return(NULL)
+  
+  if (binWidthType == "doane") {  # https://en.wikipedia.org/wiki/Histogram#Doane's_formula
+    sigma.g1 <- sqrt((6*(length(variable) - 2)) / ((length(variable) + 1)*(length(variable) + 3)))
+    g1 <- mean(abs(variable)^3)
+    k <- 1 + log2(length(variable)) + log2(1 + (g1 / sigma.g1))
+    binWidthType <- k
+  }
+  
+  if (binWidthType == "fd" & nclass.FD(variable) > 10000) # FD-method will produce extreme number of bins and crash ggplot, mention this in footnote
+    binWidthType <- 10000
+  
+  if (binWidthType == "manual")
+    binWidthType <- numberOfBins
```

Comment:
```suggestion
  } else if (binWidthType == "fd" && nclass.FD(variable) > 10000) { # FD-method will produce extreme number of bins and crash ggplot, mention this in footnote
    binWidthType <- 10000
  } else if (binWidthType == "manual") { 
    binWidthType <- numberOfBins
  }
```
`else if` + `&&` in if statements.